### PR TITLE
Add drop filters

### PR DESCRIPTION
**Pull request by mirii1994**
Add compatibility to [logzio's drop filter API](https://docs.logz.io/api/#tag/Drop-filters)

### CODE REVIEW

File: logzio/resource_drop_filter.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+		_, err = dropFilterClient(m).ActivateDropFilter(d.Id())
+	} else {
+		_, err = dropFilterClient(m).DeactivateDropFilter(d.Id())
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return resource.Retry(d.Timeout(schema.TimeoutUpdate), func() *resource.RetryError {
```
Comment (Doron-Bargo) [inline logzio/resource_drop_filter.go:141]: didnt understand why we need this

File: logzio/resource_drop_filter.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+		condition := element.(map[string]interface{})
+		conditionToAppend.FieldName = condition[dropFilterFieldName].(string)
+		conditionToAppend.Value = getValueObjectByType(condition[dropFilterValue].(string))
+		fieldConditions = append(fieldConditions, conditionToAppend)
+	}
+
+	return fieldConditions
+}
+
+func getValueObjectByType(value string) interface{} {
```
Comment (Doron-Bargo) [inline logzio/resource_drop_filter.go:217]: are you going to use this functions in more api? 
maybe worth create a utils?

File: logzio/resource_drop_filter.go [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+	returnBool, err := strconv.ParseBool(value)
+	if err == nil {
+		return returnBool
+	}
+
+	return value
+}
+
+func convertObjectToString(value interface{}) string {
```
Comment (Doron-Bargo) [inline logzio/resource_drop_filter.go:243]: are you going to use this functions in more api? 
maybe worth create a utils?

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]
Amazing work. Please see my comments

File: logzio/resource_drop_filter.go (hunk 1 above)
Comment (mirii1994) [reply logzio/resource_drop_filter.go:141]: A retry? I think it's best practice to add a retry for every resource...

**Review by Doron-Bargo — APPROVED** [review]
Great work

**Pull request merged**